app/assets/javascripts/table-select-all.es6

```diff
@@ -0,0 +1,31 @@
+$(document).ready(function() {
+  if ($('.select-clear-all').length) {
+    function resetSelectClear(table) {
+      const link = table.find('.select-clear-all');
+      link.removeClass('select-all clear-all');
+
+      if (table.find('input:checked').length) {
+        link.addClass('clear-all').text(link.data('copy-clear'));
+      }
+      else {
+        link.addClass('select-all').text(link.data('copy-select'));
+      }
+    }
+
+    $('.select-clear-all').click(function() {
+      const table = $(this).parents('table');
+      if($(this).hasClass('clear-all')) {
+        table.find('input:checked').trigger('click');
+      } else if ($(this).hasClass('select-all')) {
+        table.find('input:not(:checked)').trigger('click');
+      }
+      return false;
+    });
+
+    $('table input').click(function() {
+      resetSelectClear($(this).parents('table'))
```

Comment:
If I read this correctly, if any check box is clicked, all the check boxes are reset even though only one check box has changed.

Reply:
`resetSelectClear` doesn't touch the checboxes. It only changes the class and copy of the "select all / clear all" link. Maybe the name of that method is confusing...

Reply:
OK